digdag-server/src/main/java/io/digdag/server/rs/ProjectResource.java

```diff
@@ -157,8 +157,10 @@
     // GET  /api/projects/{id}/workflow?name=name&revision=name    # lookup a workflow of a past revision of a project by name
 
     private static final Logger logger = LoggerFactory.getLogger(ProjectResource.class);
-    private static final int ARCHIVE_TOTAL_SIZE_LIMIT = 2 * 1024 * 1024;
-    private static final int ARCHIVE_FILE_SIZE_LIMIT = ARCHIVE_TOTAL_SIZE_LIMIT;
+    private static final int DEFAULT_ARCHIVE_TOTAL_SIZE_LIMIT = 2 * 1024 * 1024;
+    private static final int DEFAULT_ARCHIVE_FILE_SIZE_LIMIT = DEFAULT_ARCHIVE_TOTAL_SIZE_LIMIT;
```

Comment:
Please tell me why `DEFAULT_ARCHIVE_FILE_SIZE_LIMIT` is necessary?

Reply:
No reason. It based on @sonots [comment](https://github.com/treasure-data/digdag/pull/994#commitcomment-32353677). but he also doesn't have any reason. Shall I remove it?

Reply:
There are two ARCHIVE_TOTAL_SIZE_LIMIT and ARCHIVE_FILE_SIZE_LIMIT parameters.
One is for the default of the first, another is for the default of the second.

Rather, any reasons that we have two parameters?

Reply:
@sonots-zozo we may want to limit bytes of one file for `DEFAULT_ARCHIVE_FILE_SIZE_LIMIT` in the future instead of total size limit. The current `ARCHIVE_FILE_SIZE_LIMIT = ARCHIVE_TOTAL_SIZE_LIMIT` doesn't make sense but, 2 parameters are fine for me. What do you think?

@hiroyuki-sato Yes, `DEFAULT_ARCHIVE_FILE_SIZE_LIMIT` could be removed at least for now. What do you think?

Reply:
@sonots-zozo , @muga Thanks

* Remove `DEFAULT_ARCHIVE_FILE_SIZE_LIMIT`
* Change `DEFAULT_ARCHIVE_TOTAL_SIZE_LIMIT` order

Reply:
@muga Do you mean, for now, we can have only 

* api.max_archive_total_size_limit
* DEFAULT_ARCHIVE_TOTAL_SIZE_LIMIT

but we still have

* MAX_ARCHIVE_TOTAL_SIZE_LIMIT = systemConfig.get("api.max_archive_total_size_limit", Integer.class, DEFAULT_ARCHIVE_TOTAL_SIZE_LIMIT);
* MAX_ARCHIVE_FILE_SIZE_LIMIT = MAX_ARCHIVE_TOTAL_SIZE_LIMIT

and we may fix in the future?

It is fine for me as long as a comment is left at `MAX_ARCHIVE_FILE_SIZE_LIMIT = MAX_ARCHIVE_TOTAL_SIZE_LIMIT`.

Reply:
@sonots-zozo 
What do you think about this modification?

```diff
diff --git a/digdag-server/src/main/java/io/digdag/server/rs/ProjectResource.java b/digdag-server/src/main/java/io/digdag/server/rs/ProjectResource.java
index 5a1e20f76f..4b15b5634a 100644
--- a/digdag-server/src/main/java/io/digdag/server/rs/ProjectResource.java
+++ b/digdag-server/src/main/java/io/digdag/server/rs/ProjectResource.java
@@ -159,6 +159,7 @@ public class ProjectResource
     private static final Logger logger = LoggerFactory.getLogger(ProjectResource.class);
     private static int MAX_ARCHIVE_TOTAL_SIZE_LIMIT;
     private static final int DEFAULT_ARCHIVE_TOTAL_SIZE_LIMIT = 2 * 1024 * 1024;
+    // `MAX_ARCHIVE_FILE_SIZE_LIMIT = MAX_ARCHIVE_TOTAL_SIZE_LIMIT` for now
     private static int MAX_ARCHIVE_FILE_SIZE_LIMIT;
     private static int MAX_SESSIONS_PAGE_SIZE;
     private static final int DEFAULT_SESSIONS_PAGE_SIZE = 100;
```

Reply:
@sonots-zozo sorry to your confusing. Yes, about `MAX_ARCHIVE_FILE_SIZE_LIMIT` (former `ARCHIVE_FILE_SIZE_LIMIT`), we want to fix it and make it configureable as well in the future to stop pushing project archive has unexpected files more earlier on digdag-server. I can leave a comments around `MAX_ARCHIVE_FILE_SIZE_LIMIT = MAX_ARCHIVE_TOTAL_SIZE_LIMIT`.

And to make the limit of archive total size configureable, these 3: `api.max_archive_total_size_limit`, `DEFAULT_ARCHIVE_TOTAL_SIZE_LIMIT`, `MAX_ARCHIVE_TOTAL_SIZE_LIMIT` are good. Users can specify the limit by `api.max_archive_total_size_limit`. If the system config is specified at bootstrap of Digdag, Digdag sets the value to `MAX_ARCHIVE_TOTAL_SIZE_LIMIT`. Otherwise, it uses `DEFAULT_ARCHIVE_TOTAL_SIZE_LIMIT` for the limit. I think that they are ok for us. Please correct me if wrong.

Reply:
@muga thanks, it is clear. It is fine for me for now.

@hiroyuki-sato 

```
// TODO: we may want to limit bytes of one file for `MAX_ARCHIVE_FILE_SIZE_LIMIT ` in the future instead of total size limit.
```

or muga-san can write about it.

Reply:
Fixed in 1edde99. Please take a look again.